aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,729 @@
+  }
+
+Comments
+^^^^^^^^^^
+
+Razor supports C# and HTML comments. The following markup:
+
+.. code-block:: none
+
+  @{
+      /* C# comment. */
+      // Another C# comment.
+  }
+  <!-- HTML comment -->
+
+Is rendered by the server as:
+
+.. code-block:: none
+
+  <!-- HTML comment -->
+
+Razor comments are removed by the server before the page is rendered. Razor uses ``@*  *@`` to delimit comments. The following code is commented out, so the server will not render any markup:
+
+.. code-block:: none
+
+    @*
+    @{
+        /* C# comment. */
+        // Another C# comment.
+    }
+    <!-- HTML comment -->
+   *@
+
+.. _Razor-Directives-label:
+
+Directives
+-----------
+Razor directives are represented by implicit expressions with reserved keywords following the ``@`` symbol. A directive will typically change the way a page is parsed or enable different functionality within your Razor page. A Razor page is just a generated C# file. A simple example of what a Razor page generates:
+
+.. literalinclude:: razor/sample/Views/Home/Contact8.cshtml
+  :language: html
+
+The Razor markup above will generate a class similar to the following:
+
+.. code-block:: c#
+
+  public class _Views_Something_cshtml : RazorPage<dynamic>
+  {
+      public override async Task ExecuteAsync()
+      {
+          var output = "Hello World";
+
+          WriteLiteral("/r/n<div>Output: ");
+          Write(output);
+          WriteLiteral("</div>");
+      }
+  }
+
+:ref:`Razor-CustomCompilationService-label` explains how to view this generated class. Understanding how Razor generates code for a view will make it easier to understand how directives work.
+
+``@using``
+^^^^^^^^^^^^^
+
+The ``@using`` directive will add the c# ``using`` directive to the generated razor page:
+
+.. literalinclude:: razor/sample/Views/Home/Contact9.cshtml
+  :language: html
+
+``@model``
+^^^^^^^^^^^^
+
+The ``@model`` directive allows you to specify the type of the model passed to your Razor page. It uses the following syntax:
+
+.. code-block:: none
+
+  @model TypeNameOfModel<TModel>
+
+For example, if you create a new ASP.NET Core MVC app with individual user accounts, the *Views/Account/Login.cshtml* Razor view file contains the follow model declaration:
+
+.. code-block:: c#
+
+  @model LoginViewModel
+
+In the class example in :ref:`Razor-Directives-label`, the class generated inherits from ``RazorPage<dynamic>``. By adding an ``@model`` you control what’s inherited. For example
+
+.. code-block:: c#
+
+  @model LoginViewModel
+
+Generates the following class
+
+.. code-block:: c#
+
+ public class _Views_Account_Login_cshtml : RazorPage<LoginViewModel>
+
+This allows you to access the strongly typed model in your Razor page through the ``Model`` property:
+
+.. code-block:: none
+
+  <div>The Login Email: @Model.Email</div>
+
+Obviously you must pass the model from the controller to the view. See :ref:`Strongly-typed-models-keyword-label` for more information.
```

Comment:
The `Strongly-typed-models-keyword-label` label isn't defined. Possible casing issue?